### DISCUSSION TITLE
Feature Request: Set max depth for lists

### DISCUSSION THREAD

**Discussion by nooblyf** [ueberdosis/tiptap#4576] [Feature Requests]
### Description

Right now there doesn't seem to be any limit on how many times you can indent a list by pressing the tab key, or simply pasting a deeply nested list in the editor

### Use Case

I think it's a good idea to have control over this, to have a setting something like `maxDepth` which prevents indenting over a certain limit and also handles cases where a deeply nested list is copy-pasted.

### Type

New feature

**Comment by alexw-at** [comment]
This would be a great feature. I want to disable all nesting and only allow "top level" lists

**Comment by bdbch** [comment]
Interesting idea - what should happen with pasted lists? Would you expect them to keep their nesting or break it and nest everything on one level?

I don't know if nesting should be broken as for most content that would lead to a loss of context when it comes to the content.

**Comment by alexw-at** [comment]
I think the nesting should be broken because otherwise there would be an invalid list in the editor.
Adding an additional `enforceMaxDepthOnPaste` setting could also be a good option for other use cases.
